### PR TITLE
Unified date, time, and datetime input widgets

### LINKED ISSUE #389: "Dates" collected by DataHarmonizer are datetime objects
see 
- https://github.com/turbomam/examples-first-cookiecutter/issues/74

### LINKED ISSUE #391: Implement time and datetime datatypes
So far DataHarmonizer only has a "date" datatype, which is treated as xsd:date under the hood.  It would be good to add a "time" and a "datetime" datatype.  

A time-only field would be implemented as xsd:time, and Handsontable has a built-in datatype "time" for that, with validation happening in the cell, as noted by @pkalita-lbl (https://github.com/cidgoh/DataHarmonizer/issues/389, https://handsontable.com/docs/7.4.2/demo-time.html ) 

The datetime datatype would be implemented as xsd:datetime, YYYY-MM-DDThh:mm:ss.  A popup UI control for that needs to be located.

### PR DESCRIPTION
**Pull request by pkalita-lbl**
* These changes add three [custom Handsontable editors](https://handsontable.com/docs/7.4.2/tutorial-cell-editor.html#page-creating-custom-editor) to handle editing [times](https://github.com/cidgoh/DataHarmonizer/pull/395/files#diff-9627ec1d5e953a82ced426ed729150b77bb5994fca20a2824aa5c826139b48cd), [dates](https://github.com/cidgoh/DataHarmonizer/pull/395/files#diff-dd991a0d40d19f3df4214a80ba8520dd9a1a2d96b36f55b001bd2c7b8c9f7d5b), and [datetimes](https://github.com/cidgoh/DataHarmonizer/pull/395/files#diff-9de1e508c6accc41fe9bf46805e9ddf63d2cb80ca07a79e02fd78d1829854625). Each inherits from one [editor superclass](https://github.com/cidgoh/DataHarmonizer/pull/395/files#diff-4cde2414eba67885752fa2487edda9cdcc34bd170ac7231c6ce647928400bd9f) that handles setting up a [Flatpickr](https://flatpickr.js.org/) instance and hooking it up to the necessary Handsontable events. The concrete subclasses are just responsible for providing Flatpickr configuration. I chose Flatpickr over native browser inputs because the native controls generally lack the ability to control the [formatting](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/datetime-local#value) of the selected date. These editors are registered as part of three [custom cell types](https://github.com/cidgoh/DataHarmonizer/pull/395/files#diff-63a4bd3b333612153ec2f0fcc43afe720061f4b3c0384ba2d9190629b2c9d7d9R42-R53) and [applied to columns](https://github.com/cidgoh/DataHarmonizer/pull/395/files#diff-63a4bd3b333612153ec2f0fcc43afe720061f4b3c0384ba2d9190629b2c9d7d9R1098-R1112) based on the column datatype. The format of the values as displayed in the Handsontable grid are controlled by [three new options](https://github.com/cidgoh/DataHarmonizer/pull/395/files#diff-63a4bd3b333612153ec2f0fcc43afe720061f4b3c0384ba2d9190629b2c9d7d9R83-R85) that can be passed to the `DataHarmonizer` constructor. Acceptable values for these options should be based on the [Unicode standard](https://www.unicode.org/reports/tr35/tr35-dates.html#Date_Field_Symbol_Table) date and time tokens. Fixes #391 
* These changes also make the utility functions `dataArrayToObject` and `dataObjectToArray` used in JSON import/export aware of the `DataHarmonizer` instance's date and time formats and provide finer control over how to serialize/deserialize dates. This is ultimately controlled by [another new](https://github.com/cidgoh/DataHarmonizer/pull/395/files#diff-63a4bd3b333612153ec2f0fcc43afe720061f4b3c0384ba2d9190629b2c9d7d9R86) `DataHarmonizer` constructor option. The default is that exported JSON will have dates in a format that is compatible with [JSON Schema validation](https://json-schema.org/understanding-json-schema/reference/string.html#dates-and-times). Other [possible options](https://github.com/cidgoh/DataHarmonizer/pull/395/files#diff-92bbac9a308cd5fcf9db165841f2d90ce981baddcb2b1e26cfff170929af3bd1R4) are having dates serialized according to the [default serialization of Date objects](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/toJSON) or in the same format as they're displayed in the Handsontable grid. Fixes #389.

### CODE REVIEW

**Comment by pkalita-lbl** [comment]
@ddooley I think this read for review now. Please let me know if you'd like me to walk you through any of it!

File: tests/fields.test.js [hunk 1]
```diff
@@ -68,18 +78,41 @@ describe('dataArrayToObject', () => {
...
     expect(dataObject).toEqual({
-      a: '5.85',
-      b: '21.25',
-      f: ['5', 'asdf', '18'],
+      b: 21.25,
+      f: [5, 18],
+    });
+  });
+
+  test('returns correctly formatted string with dateBehavior = JSON_SCHEMA_FORMAT', () => {
```
Comment (ddooley) [inline tests/fields.test.js:98]: I see now the JSON_SCHEMA_FORMAT has to be defined in order to handle conversion from internal Javascript date object out to JSON (using XML standard date/time types).  And then you allow custom formats too for date, datetime, and time.

File: lib/utils/fields.js [hunk 2]
```diff
@@ -40,14 +47,67 @@ export function fieldUnitBinTest(fields, col) {
...
+    timeFormat,
+  });
+  const formatDatesAndTimes = (dateObject, datatype) => {
+    if (!isDate(dateObject)) {
+      return dateObject;
+    }
+    if (dateBehavior === DATE_OBJECT) {
+      return dateObject;
+    }
+    if (dateBehavior === INPUT_FORMAT) {
```
Comment (ddooley) [inline lib/utils/fields.js:87]: And I gather that the INPUT_FORMAT is all about the way date/time picker fields format and show dates/times in HandsonTable cells on screen, when they initially were read in from a json data file into javascript objects?!

**Review by ddooley — APPROVED** [review]
Seems like this successfully dived into making distinction between date/time as it appears in DataHarmonizer cells with respect to date and time picker, or manual input; vs conversion from loaded json file, and writing from javascript objects to json format file.

**Pull request merged**